xarray/core/formatting_html.py

```diff
@@ -341,3 +343,130 @@ def dataset_repr(ds) -> str:
     ]
 
     return _obj_repr(ds, header_components, sections)
+
+
+def summarize_datatree_children(children: Mapping[str, Any]) -> str:
+    N_CHILDREN = len(children) - 1
+
+    # Get result from datatree_node_repr and wrap it
+    lines_callback = lambda n, c, end: _wrap_datatree_repr(
+        datatree_node_repr(n, c), end=end
+    )
+
+    children_html = "".join(
+        (
+            lines_callback(n, c, end=False)  # Long lines
+            if i < N_CHILDREN
+            else lines_callback(n, c, end=True)
+        )  # Short lines
+        for i, (n, c) in enumerate(children.items())
+    )
+
+    return "".join(
+        [
+            "<div style='display: inline-grid; grid-template-columns: 100%'>",
+            children_html,
+            "</div>",
+        ]
+    )
+
+
+children_section = partial(
+    _mapping_section,
+    name="Groups",
+    details_func=summarize_datatree_children,
+    max_items_collapse=1,
+    expand_option_name="display_expand_groups",
+)
+
+
+def datatree_node_repr(group_title: str, dt: Any) -> str:
```

Comment:
Looking at the original `formatting_html.py`, it looks like you inherited the `Any` typing in this file. This seems like a good time to tighten them up; I'm guessing they should probably be `DataTree` - I'll comment on the other two that I found in this file.

Reply:
Thanks for noticing this! Just pushed up a new commit with these updates